### ISSUE TITLE
Drop automated dev/stage/prod detection code in favor of a menu

### ISSUE THREAD

**Issue by ghost** [mozilla/testpilot#1198]
When Test Pilot starts, there is some code that attempts to look into your history and see the most recently visited test pilot site, and then set the Test Pilot add-on to work with that site (dev/stage/prod/local).  That code might be causing some weird issues (like the hard-to-reproduce white pages on load) but it definitely affects things like #1162.  

Instead of that code, let's do a simpler check:  On startup and each time the Test Pilot menu button is clicked, look for a pref called `extensions.testpilot-addon.developermenu` and if it exists add a quick-change option to our menu button.  Like so, only, you know, prettier:

![devmenu](https://cloud.githubusercontent.com/assets/16946141/17448263/41a3d34c-5b08-11e6-9b14-89e72f5609a6.png)

Then everyone installs like normal from wherever, and if they need the menu they go to about:config and add the pref.  If you were feeling fancy, you could even put a little local/dev/stage badge on the icon, but that might be too much. :)

Thoughts?

**Comment by chuckharmston** [comment]
Duping to https://github.com/mozilla/testpilot/issues/1196, but it's a strong idea.

**Comment by lmorchard** [comment]
Yeah, the problem with this is that we need the add-on to know what environment to use before the toolbar button or doorhanger are set up.

When you click "install" from the web, the add-on gets installed & starts up. If the current configured environment matches the current web page, the add-on injects a page-mod on that page. After that, the web frontend can detect the add-on's presence, stops animating "Installing", and redirects to /experiments/.

So, tl;dr: If you don't have the proper environment set _before_ doing all the setup work in the addon (including toolbar button), the frontend web installation process never completes.
